packages/rancher-logging/overlay/templates/loggings/eks/logging.yaml

```diff
@@ -16,10 +16,15 @@ spec:
       Tag: "eks"
       Path: "/var/log/messages"
       Parser: "syslog"
-    {{ with .Values.fluentbit_tolerations }}
+    {{ $total_tolerations := concat (.Values.tolerations) (.Values.fluentbit_tolerations) }}
+    {{ with $total_tolerations }}
```

Comment:
Does the yaml turn out normally and spaced correctly when using concat ? I had assumed we would do a if statement that surround `tolerations:` and then two with statements. But this can work too.

Reply:
I had started the ```if``` route, but opted to use ```concat``` after hitting a few errors and asking folks in the Kubernetes slack.

The indentation and spacing is as expected, but there are extra newlines. This does not seem to be related to using ```concat``` though. Here is some YAML from a ```--dry-run --debug``` install...

```yaml
  fluentbit:
    image:
      repository: rancher/fluent-fluent-bit
      tag: 1.5.4
    
    
    tolerations:
      
      - effect: NoSchedule
        key: cattle.io/os
        operator: Equal
        value: linux
      - effect: NoSchedule
        key: node-role.kubernetes.io/controlplane
        value: "true"
      - effect: NoExecute
        key: node-role.kubernetes.io/etcd
        value: "true"
    
    
    nodeSelector:
      
      kubernetes.io/os: linux
```

At first, I thought this was because I used ```concat```, but the ```fluentd``` section actually doesn't use it. The ```fluentd``` section is the same as the deployment below.

```yaml    
  fluentd:
    image:
      repository: rancher/banzaicloud-fluentd
      tag: v1.11.2-alpine-2
    configReloaderImage:
      repository: rancher/jimmidyson-configmap-reload
      tag: v0.2.2
    disablePvc: true
    
    tolerations:
      
      - effect: NoSchedule
        key: cattle.io/os
        operator: Equal
        value: linux
    
    
    nodeSelector:
      
      kubernetes.io/os: linux
```

Confusingly, the deployment YAML is fine.

```yaml
      containers:
        - name: rancher-logging
          image: "rancher/banzaicloud-logging-operator:3.6.0"
          imagePullPolicy: IfNotPresent
          resources:
            {}
          ports:
            - name: http
              containerPort: 8080
      nodeSelector:
        kubernetes.io/os: linux
      tolerations:
        - effect: NoSchedule
          key: cattle.io/os
          operator: Equal
          value: linux
```

I can dive deeper into this. It's functional, but may need tidying if deemed necessary.

Reply:
Hm... I actually think I might've found the issue. The indentation of the templating lines might be creating the newlines. Testing locally... I'll update this thread.

Reply:
Might just be the helm template braces not using a minus sign `{{-`

Reply:
Found it. Need to prepend templating conditionals with a dash character. The newlines do not appear now. Updating accordingly.

Reply:
Fixed